source/material_model/reactive_fluid_transport.cc

```diff
@@ -50,6 +50,96 @@ namespace aspect
+      std::vector<double> csat_gabbro_poly_coeffs {-0.0176673, 0.0893044, 1.52732};
+      std::vector<double> Td_gabbro_poly_coeffs {-1.72277, 20.5898, 637.517};
+
+      std::vector<double> LR_MORB_poly_coeffs {-1.78177, 7.50871, -10.4840, 5.19725, 7.96365};
+      std::vector<double> csat_MORB_poly_coeffs {0.0102725, -0.115390, 0.324452, 1.41588};
+      std::vector<double> Td_MORB_poly_coeffs {-3.81280, 22.7809, 638.049};
+
+      std::vector<double> LR_sediment_poly_coeffs {-2.03283, 10.8186, -21.2119, 18.3351, -6.48711, 8.32459};
+      std::vector<double> csat_sediment_poly_coeffs {-0.150662, 0.301807, 1.01867};
+      std::vector<double> Td_sediment_poly_coeffs {2.83277, -24.7593, 85.9090, 524.898};
+
+      std::vector<std::vector<double>> LR_all_poly_coeffs {LR_peridotite_poly_coeffs, LR_gabbro_poly_coeffs, \
+                                                            LR_MORB_poly_coeffs, LR_sediment_poly_coeffs
+                                                           };
+
+      std::vector<std::vector<double>> csat_all_poly_coeffs {csat_peridotite_poly_coeffs, csat_gabbro_poly_coeffs, \
+                                                              csat_MORB_poly_coeffs, csat_sediment_poly_coeffs
+                                                             };
+
+      std::vector<std::vector<double>> Td_all_poly_coeffs {Td_peridotite_poly_coeffs, Td_gabbro_poly_coeffs, \
+                                                            Td_MORB_poly_coeffs, Td_sediment_poly_coeffs
+                                                           };
+
+      // Create arrays that will store the values of the polynomials at the current pressure
+      std::vector<double> LR_values {0, 0, 0, 0};
+      std::vector<double> csat_values {0, 0, 0, 0};
+      std::vector<double> Td_values {0, 0, 0, 0};
+
+      // Loop over the four rock types (peridotite, gabbro, MORB, sediment) and the polynomial
+      // coefficients to fill the vectors defined above. The polynomials for LR are defined in
+      // equations 13, B2, B10, and B18. csat polynomials are defined in equations 14, B1, B9, and B17.
+      // Td polynomials are defined in equations 15, B3, B11, and B19.
+      for (unsigned int i = 0; i<LR_all_poly_coeffs.size(); ++i)
+        for (unsigned int j = 0; j<LR_all_poly_coeffs[i].size(); ++j)
+          {
+            LR_values[i] += LR_all_poly_coeffs[i][j] * std::pow(inverse_pressure, LR_all_poly_coeffs[i].size() - 1 - j);
+          }
+
+      for (unsigned int i = 0; i<csat_all_poly_coeffs.size(); ++i)
+        for (unsigned int j = 0; j<csat_all_poly_coeffs[i].size(); ++j)
+          {
+            csat_values[i] += i==3 ? csat_all_poly_coeffs[i][j] * std::pow(std::log10(pressure), csat_all_poly_coeffs[i].size() - 1 - j) :\
+                              csat_all_poly_coeffs[i][j] * std::pow(pressure, csat_all_poly_coeffs[i].size() - 1 - j);
+          }
+
+      for (unsigned int i = 0; i<Td_all_poly_coeffs.size(); ++i)
```

Comment:
As I understand it, this outer loop is the same for all three parameters. (It always has length 4). 
So you can make the code shorter by doing the outer loop only once:
```suggestion
      for (unsigned int i = 0; i<LR_all_poly_coeffs.size(); ++i)
      {
        for (unsigned int j = 0; j<LR_all_poly_coeffs[i].size(); ++j)
          {
            LR_values[i] += LR_all_poly_coeffs[i][j] * std::pow(inverse_pressure, LR_all_poly_coeffs[i].size() - 1 - j);
          }

        for (unsigned int j = 0; j<csat_all_poly_coeffs[i].size(); ++j)
          {
            csat_values[i] += i==3 ? csat_all_poly_coeffs[i][j] * std::pow(std::log10(pressure), csat_all_poly_coeffs[i].size() - 1 - j) :\
                              csat_all_poly_coeffs[i][j] * std::pow(pressure, csat_all_poly_coeffs[i].size() - 1 - j);
          }

```